scripts/utils/price-utils.js

```diff
@@ -134,9 +137,48 @@ module.exports = function(web3 = web3, artifacts = artifacts) {
     return true
   }
 
+  /**
+   * Computes the amount of output token units from their price and the amount of input token units
+   * Note that the price is expressed in terms of tokens, while the amounts are in terms of token units
+   * @param {number} price amount of output token in exchange for one input token
+   * @param {BN} targetTokenAmount amount of token units that are exchanged at price
+   * @param {integer} targetTokenDecimals number of decimals of the input token
+   * @param {integer} stableTokenDecimals number of decimals of the output token
+   * @return {BN} amount of output token units obtained
+   */
+  const getOutputAmountFromPrice = function(price, targetTokenAmount, targetTokenDecimals, stableTokenDecimals) {
+    const priceFraction = Fraction.fromNumber(price)
+    const unitPriceFraction = priceFraction.mul(
+      new Fraction(new BN(10).pow(new BN(stableTokenDecimals)), new BN(10).pow(new BN(targetTokenDecimals)))
+    )
+    const stableTokenAmountFraction = unitPriceFraction.mul(new Fraction(targetTokenAmount, 1))
+    return stableTokenAmountFraction.toBN()
+  }
+
+
+  /**
+   * Computes the stable and target token amounts needed to set up an unlimited order in the exchange
+   * @param {number} price amount of stable tokens in exchange for one target token
+   * @param {integer} targetTokenDecimals number of decimals of the target token
+   * @param {integer} stableTokenDecimals number of decimals of the stable token
+   * @return {BN[2]} amounts of stable token and target token for an unlimited order at the input price
+   */
+  const getUnlimitedOrderAmounts = function(price, targetTokenDecimals, stableTokenDecimals) {
+    let targetTokenAmount = max128.clone()
+    let stableTokenAmount = getOutputAmountFromPrice(price, targetTokenAmount, targetTokenDecimals, stableTokenDecimals)
+    if (stableTokenAmount.gt(targetTokenAmount)) {
+      stableTokenAmount = max128.clone()
+      targetTokenAmount = getOutputAmountFromPrice(1/price, stableTokenAmount, stableTokenDecimals, targetTokenDecimals)
+      assert(stableTokenAmount.gte(targetTokenAmount), "Error: unable to create unlimited order")
```

Comment:
I can not imagine any case where this would throw an error!

Reply:
I also think this shouldn't ever throw, it's there just in case there was some very unexpected rounding issue since I use `1/price` instead of `price`. This is also why I added `1+Number.EPSILON` and `1-Number.EPSILON` to the test cases, to be sure that nothing like that was going on.

Reply:
I am wondering if we sometimes switch the nothign of "stableToken" and "targetToken" like, for example, when computing the bracket orders. If this is the case, then I am beginning to think that sometimes the "stableToken" is being passed into this function as the `targetToken`and the names have become misleading. If this is the case, would we agree that this naming convention could be improved? I wouldn't know what to suggest, but I suppose if there are always only two tokens and the calculations are all made relative to each other then the `referenceToken` is always stable relative to the `targetToken`. Could these be a more name pair?

Reply:
I agree that many functions in this repo are prone to having stable and target token mixed up. As I see it, the main discriminant between "stable" and "target" is which price you are using: the price `p` always means `1` target token in exchange for `p` stable tokens. I'm not sure how to improve on the current naming system, though. Never give the price alone, but have a "price triplet" with price, "stable" and "target"? But what when there is no price?